src/Controls/tests/DeviceTests/Elements/HybridWebView/HybridWebViewTests.cs

```diff
@@ -79,5 +82,279 @@ public async Task LoadsHtmlAndSendReceiveRawMessage()
+					HeightRequest = 100,
+
+					HybridRoot = "HybridTestRoot",
+				};
+
+				var handler = CreateHandler(hybridWebView);
+
+				var platformView = handler.PlatformView;
+
+				// Set up the view to be displayed/parented and run our tests on it
+				await AttachAndRun(hybridWebView, async (handler) =>
+				{
+					await WaitForHybridWebViewLoaded(hybridWebView);
+
+					var x = 123.456m;
+					var y = 654.321m;
+
+					var result = await hybridWebView.InvokeJavaScriptAsync<decimal>(
+						"EvaluateMeWithParamsAndReturn",
+						HybridWebViewTestContext.Default.Decimal,
+						new object[] { x, y },
+						new[] { HybridWebViewTestContext.Default.Decimal, HybridWebViewTestContext.Default.Decimal });
+
+					Assert.Equal(777.777m, result);
+				});
+			});
+		}
+
+		[Fact]
+		public async Task InvokeJavaScriptMethodWithParametersAndComplexResult()
+		{
+#if ANDROID
+			// NOTE: skip this test on older Android devices because it is not currently supported on these versions
+			if (!System.OperatingSystem.IsAndroidVersionAtLeast(24))
+			{
+				return;
+			}
+#endif
+
+			SetupBuilder();
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				var hybridWebView = new HybridWebView()
+				{
+					WidthRequest = 100,
+					HeightRequest = 100,
+
+					HybridRoot = "HybridTestRoot",
+				};
+
+				var handler = CreateHandler(hybridWebView);
+
+				var platformView = handler.PlatformView;
+
+				// Set up the view to be displayed/parented and run our tests on it
+				await AttachAndRun(hybridWebView, async (handler) =>
+				{
+					await WaitForHybridWebViewLoaded(hybridWebView);
+
+					var x = 123.456m;
+					var y = 654.321m;
+
+					var result = await hybridWebView.InvokeJavaScriptAsync<ComputationResult>(
+						"AddNumbers",
+						HybridWebViewTestContext.Default.ComputationResult,
+						new object[] { x, y },
+						new[] { HybridWebViewTestContext.Default.Decimal, HybridWebViewTestContext.Default.Decimal });
+
+					Assert.NotNull(result);
+					Assert.Equal(777.777m, result.result);
+					Assert.Equal("Addition", result.operationName);
+				});
+			});
+		}
+
+		[Fact]
+		public async Task InvokeAsyncJavaScriptMethodWithParametersAndComplexResult()
+		{
+#if ANDROID
+			// NOTE: skip this test on older Android devices because it is not currently supported on these versions
+			if (!System.OperatingSystem.IsAndroidVersionAtLeast(24))
+			{
+				return;
+			}
+#endif
+
+			SetupBuilder();
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				var hybridWebView = new HybridWebView()
+				{
+					WidthRequest = 100,
+					HeightRequest = 100,
+
+					HybridRoot = "HybridTestRoot",
+				};
+
+				var handler = CreateHandler(hybridWebView);
+
+				var platformView = handler.PlatformView;
+
+				// Set up the view to be displayed/parented and run our tests on it
+				await AttachAndRun(hybridWebView, async (handler) =>
+				{
+					await WaitForHybridWebViewLoaded(hybridWebView);
+
+					// TODO: This EvaluateMeWithParamsAndAsyncReturn
```

Comment:
You saw this yesterday, but just as a reminder to remove later!

Reply:
Thanks for finding these - I'll fix in a follow-up PR to avoid restarting the CI for this.

Reply:
Actually fixed now because I'm rebasing the PR anyway.